src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/XmlDescriptor.php

```diff
@@ -340,6 +341,23 @@ private function getContainerDefinitionDocument(Definition $definition, $id = nu
             $serviceXML->setAttribute('factory-method', $definition->getFactoryMethod());
         }
 
+        if ($factory = $definition->getFactory()) {
+            $serviceXML->appendChild($factoryXML = $dom->createElement('factory'));
+
+            if (is_array($factory)) {
+                if ($factory[0] instanceof Reference) {
+                    $factoryXML->setAttribute('service', (string) $factory[0]);
+                } elseif ($factory[0] instanceof Definition) {
+                    // XXX What should we put here?
```

Comment:
Are you sure or can you provide an example? At least we don't have such an example [in the documentation](http://symfony.com/doc/current/components/dependency_injection/factories.html).